src/main/java/ui/components/NavigableListView.java

```diff
@@ -111,25 +114,36 @@ public void restoreSelection() {
                 selectedIndex = Optional.of(nextIndex);
 
                 // The next item will be considered selected
-                onItemSelected.accept(nextIndex);
+                setItemSelectedParams(nextIndex, SHOULD_LOAD_BVIEW_ISSUE);
             }
         }
     }
 
+    /**
+     * Passes the relevant index and boolean value for whether the bview should be loaded
+     * to the callback variable onItemSelected
+     */
+    private void setItemSelectedParams(int index, boolean shouldLoadBViewIssue){
+        onItemSelected.accept(index, shouldLoadBViewIssue);
+    }
+
     public abstract boolean areItemsEqual(T item1, T item2);
 
     private void setupMouseEvents() {
         setOnMouseClicked(e -> {
             int currentlySelected = getSelectionModel().getSelectedIndex();
-
             // The currently-selected index is sometimes -1 when an issue is clicked.
             // When this happens we ignore this event.
-
             if (currentlySelected != -1) {
                 selectedIndex = Optional.of(currentlySelected);
 
                 logger.info("Mouse click on item index " + selectedIndex.get());
-                onItemSelected.accept(selectedIndex.get());
+                // whenever the right mouse button is clicked, do not load issue on bView.
```

Comment:
I don't think this comment is necessary. Same reason as above: this class doesn't know about the browser or how callers are going to use it.